paddle/operators/conv_op.h

```diff
@@ -91,32 +91,28 @@ class GemmConvKernel : public framework::OpKernel<T> {
 
     const int batch_size = static_cast<int>(input->dims()[0]);
 
-    // filter_shape_vec: {k_h, k_w} or {k_d, k_h, k_w}
+    // filter_shape_vec: {k_o, k_i, k_h, k_w} or {k_o, k_i, k_d, k_h, k_w}
     std::vector<int64_t> filter_shape_vec(framework::vectorize(filter.dims()));
-    filter_shape_vec.erase(filter_shape_vec.begin(),
-                           filter_shape_vec.begin() + 2);
-
-    // output_shape_vec: {o_h, o_w} or {o_d, o_h, o_w}
+    // output_shape_vec: {o_n, o_c, o_h, o_w} or {o_n, o_c, o_d, o_h, o_w}
     std::vector<int64_t> output_shape_vec(framework::vectorize(output->dims()));
-    output_shape_vec.erase(output_shape_vec.begin(),
-                           output_shape_vec.begin() + 2);
 
     // use col_shape in the im2col calculation
     // col_shape_vec: {i_c/g, k_h, k_w, o_h, o_w} or {i_c/g, k_d, k_h, k_w, o_d,
     // o_h, o_w}
-    std::vector<int64_t> col_shape_vec;
-    col_shape_vec.push_back(input->dims()[1] / groups);
-    col_shape_vec.insert(col_shape_vec.end(), filter_shape_vec.begin(),
+    std::vector<int64_t> col_shape_vec(filter_shape_vec.size() +
+                                       output_shape_vec.size() - 3);
+    col_shape_vec.assign(1, input->dims()[1] / groups);
```

Comment:
`col_shape_vec[0] = input->dims()[1] / groups` should be fine.

Reply:
It doesn't get what we want.

```
  std::vector<int> vec_i({ 501, 502, 503 });

  std::vector<int> vec (8);
  vec[0]  = 12;
  vec.insert(vec.begin()+1, vec_i.begin(), vec_i.end());
```
If we write it this way, `vec` is `{12 501 502 503 0 0 0 0 0 0 0}`.
But if we write it the other way, `vec` is `{12 501 502 503}`.
```
  std::vector<int> vec (8);
  vec.assign(1, 12);
  vec.insert(vec.begin()+1, vec_i.begin(), vec_i.end());
```